### ISSUE TITLE
Fix Windows Build

### ISSUE THREAD

**Issue by thedrow** [celery/kombu#1035]
Following the release of vine 5.0.0a1 and py-amqp 5.0.0a1 the AppVeyor build started failing.

This happened since we were using `python setup.py test` in that environment instead of tox.
That command installs dependencies using `easy_install` which apparently does not implement [PEP-440](https://www.python.org/dev/peps/pep-0440/). Therefore, it picks the latest available version although it is an alpha version unlike pip which requires the `--pre` option to install alpha versions.

An attempt was made to migrate that build environment to tox, but tox installs our extra dependencies such as pycurl which does not release wheels for Windows.

We can either:

- Create a tox environment for Windows which avoids installing any extras.
- Work our way to resolve each and every build issue of our dependencies.

**Comment by matteius** [comment]
This is an area of interest to me as I run windows 10 and us VirtualBox to run ubuntu images for 99% of my development efforts these days.   I even see I have some stashed billiard changes to try and debug some issues related to this a while back when I saw the appveyor builds always failing.   There are some areas I think could be improved for running in a windows environment and it is easy for me to test some of that based on my setup, but it is difficult since the automated CI is always broken around that.  

While it may be possible to get to the second bullet of "Work our way to resolve each and every build issue of our dependencies." it may be useful to do the first thing of "Create a tox environment for Windows which avoids installing any extras." so that we can actually have a clean environment to get to that second part.

**Comment by thedrow** [comment]
We probably need to remove curl from Windows CI altogether until pycurl/pycurl#569 is resolved.

**Comment by mpuhacz** [comment]
> We probably need to remove curl from Windows CI altogether until [pycurl/pycurl#569](https://github.com/pycurl/pycurl/issues/569) is resolved.

It's also possible to install libcurl from a prebuilt wheel. Unfortunately, I don't have access to a windows machine.
https://www.lfd.uci.edu/~gohlke/pythonlibs/#pycurl

**Comment by matteius** [comment]
I have spent some hours on this looking at curl and libcurl from different angles.  I got pretty far with hacking through the setup.py and winbuild.py in pycurl but still ran into downstream dependencies with libcurl not being in the paths I searched on appveyor (though I did not check every path) it was a long hack to get a build to go this far:  https://ci.appveyor.com/project/ask/kombu/builds/26551748/job/3tt1881v9aggvvyu

So while I agree that if the windows wheel just existed for pycurl we wouldn't have the libcurl problems on compilation of that library when the setup.py installs.  However in studying the kombu usage of pycurl I came to a more fundamental question of why curl and why not python requests?

I tried to answer my own question, in kombu there is the code from Ask in 2014 adding the CurlClient and it has some _handle_socket method that appears to never be called, and the actual usages of it seem confined to the AWS connection.py with classes such as AsyncHTTPSConnection, it seems all of the classes in there are just using normal HTTP verbs -- is there any reason why I should not be trying to convert that class to use python requests?  I already started down this path and having spent less time than trying to get pycurl to build on windows, it seems like a promising direction to head in unless I am missing something that curl gets us that requests could not.
